src/sentry/relay/config/metric_extraction.py

```diff
@@ -113,14 +111,19 @@ def get_metric_extraction_config(
     sentry_sdk.set_tag("organization_id", project.organization_id)
 
     with sentry_sdk.start_span(op="get_on_demand_metric_specs"):
-        alert_specs, widget_specs = get_on_demand_metric_specs(timeout, project)
+        on_demand_specs = run_experimental_config_builder(get_on_demand_metric_specs, project)
+        if on_demand_specs is not None:
+            # unpacking the tuple returned from get_on_demand_metric_specs
+            alert_specs, widget_specs = on_demand_specs
+        else:
+            # exception occurred during the get_on_demand_metric_specs run
+            alert_specs, widget_specs = [], []
     with sentry_sdk.start_span(op="generate_span_attribute_specs"):
         span_attr_specs = _generate_span_attribute_specs(project)
```

Comment:
At least `_generate_span_attribute_specs` should also be wrapped by our guarding function, as it accesses the database and might lead to similar slowdowns as the on-demand function.